auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -1358,4 +1654,26 @@ public XMLStreamWriter getXmlStreamWriter() {
             return xmlStreamWriter;
         }
     }
+
+    public static class IntegrityResult {
+        private final boolean valid;
+        private final ArrayList<Identity> identities;
```

Comment:
Just wondering if this could be an `ArrayList<String>` instead? That way, you'd just build up the list while verifying integrity and wouldn't need to iterate over the identities in `getIdentityNames`.

Reply:
I was just thinking if in the future this was to be used for another purpose and another property of the identity was required, it would be useful. Although I could change it to `ArrayList<String> identityNames` for this use case, and if the identities are required in the future, the identities arraylist could be added back

Reply:
Yeah, I think we just need the string names for now.

Reply:
Updated